cassiopeia/solver/SpectralSolver.py

```diff
@@ -119,19 +123,20 @@ def perform_split(
         cuts needed to be explored.
 
         Args:
-            samples: A list of samples, represented as integer indices
+            samples: A list of samples, represented by their string names
 
         Returns:
             A tuple of lists, representing the left and right partition groups
         """
+        int_samples = list(map(lambda x: self.name_to_index[x], samples))
 
         G = graph_utilities.construct_similarity_graph(
             self.unique_character_matrix,
             self.missing_char,
-            samples,
+            int_samples,
```

Comment:
Would it be better here to pass in actual sample names to `construct_similarity_graph`? 

I feel like generally we're actually making these function _less_ usable, in favor of efficiency. I personally think that the conversion to integer indices and numpy arrays should occur behind the scenes, all the time, so that regular users aren't confused about how to interface with these function. It's a similar point to how we'd like to interface with the `calculate_mutation_frequencies` function.